src/mtype.d

```diff
@@ -7963,13 +7948,14 @@ public:
                 return oe;
             return new DotExp(e.loc, e, oe);
         }
+
         Declaration d = s.isDeclaration();
-        debug
+        if (!d)
         {
-            if (!d)
-                printf("d = %s '%s'\n", s.kind(), s.toChars());
+            e.error("%s.%s is not a declaration", e.toChars(), ident.toChars());
```

Comment:
My concern is: different from c-dmd, assert is useless with **released** d-dmd. Currently it's compiled with -release, so assert is never tested. Therefore if `d` would become `null`, it would cause OS-specific segmentation fault. The added error message is definitely better than that.